client/web/web.go

```diff
@@ -880,6 +1026,23 @@ func (s *Server) serveGetExitNodes(w http.ResponseWriter, r *http.Request) {
 	writeJSON(w, exitNodes)
 }
 
+// maskedPrefs is the subset of ipn.MaskedPrefs that are
+// allowed to be editable via the web UI.
+type maskedPrefs struct {
+	RunSSHSet bool
+	RunSSH    bool
+}
+
+func (s *Server) serveUpdatePrefs(ctx context.Context, prefs maskedPrefs) error {
```

Comment:
The `/local/v0/prefs` endpont was previously guarded by a [content-type check](https://github.com/tailscale/tailscale/blob/main/client/web/web.go#L1101-L1107) when it was going through `proxyRequestTolocalAPI`. Do we want to add that in here as well to maintain parity with the previous behaviour?

Reply:
Oh yes, we'll need that, will add